docs/environment-variables.md
# General environment variables

| Name (a star<b>\*</b> means the variable is required) | Description                                                                                                                                                                                                                             | Default value                |
| ----------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------- |
| HASURA_GRAPHQL_JWT_SECRET<b>\*</b>                    | Key used for generating JWTs. Must be `HMAC-SHA`-based and the same as configured in Hasura. [More info](https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt.html#running-with-jwt)                                      |                              |
| HASURA_GRAPHQL_DATABASE_URL<b>\*</b>                  | [PostgreSQL connection URI](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING). Required to inject the `auth` schema into the database.                                                                       |                              |
| HASURA_GRAPHQL_GRAPHQL_URL<b>\*</b>                   | Hasura GraphQL endpoint. Required to manipulate account data. For instance: `https://graphql-engine:8080/v1/graphql`                                                                                                                    |                              |
| HASURA_GRAPHQL_ADMIN_SECRET<b>\*</b>                  | Hasura GraphQL Admin Secret. Required to manipulate account data.                                                                                                                                                                       |                              |
| AUTH_HOST                                             | Server host. This option is available until Hasura-auth `v0.6.0`. [Docs](http://expressjs.com/en/5x/api.html#app.listen)                                                                                                                | `0.0.0.0`                    |
| AUTH_PORT                                             | Server port. [Docs](http://expressjs.com/en/5x/api.html#app.listen)                                                                                                                                                                     | `4000`                       |
| AUTH_SERVER_URL                                       | Server URL of where Hasura Backend Plus is running. This value is to used as a callback in email templates and for the OAuth authentication process.                                                                                    |                              |
| AUTH_CLIENT_URL                                       | URL of your frontend application. Used to redirect users to the right page once actions based on emails or OAuth succeed.                                                                                                               |                              |
| AUTH_CONCEAL_ERRORS                                   | Conceal sensitive error messages to avoid leaking information about user accounts to attackers                                                                                                                                          | `false`                      |
| AUTH_SMTP_HOST                                        | SMTP server hostname used for sending emails                                                                                                                                                                                            |                              |
| AUTH_SMTP_PORT                                        | SMTP port                                                                                                                                                                                                                               | `587`                        |
| AUTH_SMTP_USER                                        | Username to use to authenticate on the SMTP server                                                                                                                                                                                      |                              |
| AUTH_SMTP_PASS                                        | Password to use to authenticate on the SMTP server                                                                                                                                                                                      |                              |
| AUTH_SMTP_SENDER                                      | Email to use in the `From` field of the email                                                                                                                                                                                           |                              |
| AUTH_SMTP_AUTH_METHOD                                 | SMTP authentication method                                                                                                                                                                                                              | `PLAIN`                      |
| AUTH_SMTP_SECURE                                      | Enables SSL. [More info](https://nodemailer.com/smtp/#tls-options).                                                                                                                                                                     | `false`                      |
| AUTH_GRAVATAR_ENABLED                                 |                                                                                                                                                                                                                                         | `true`                       |
| AUTH_GRAVATAR_DEFAULT                                 | One of '404', 'mp', 'identicon', 'monsterid', 'wavatar', 'retro', 'robohash', 'blank'.                                                                                                                                                  | `blank`                      |
| AUTH_GRAVATAR_RATING                                  | One of 'g', 'pg', 'r', 'x'.                                                                                                                                                                                                             | `g`                          |
| AUTH_ANONYMOUS_USERS_ENABLED                          | Enables users to register as an anonymous user.                                                                                                                                                                                         | `false`                      |
| AUTH_DISABLE_NEW_USERS                                | If set, new users will be disabled after finishing registration and won't be able to connect.                                                                                                                                           | `false`                      |
| AUTH_ACCESS_CONTROL_ALLOWED_EMAILS                    | Comma-separated list of emails that are allowed to register.                                                                                                                                                                            |                              |
| AUTH_ACCESS_CONTROL_ALLOWED_EMAIL_DOMAINS             | Comma-separated list of email domains that are allowed to register. If `ALLOWED_EMAIL_DOMAINS` is `tesla.com,ikea.se`, only emails from tesla.com and ikea.se would be allowed to register an account.                                  | `` (allow all email domains) |
| AUTH_ACCESS_CONTROL_BLOCKED_EMAILS                    | Comma-separated list of emails that cannot register.                                                                                                                                                                                    |                              |
| AUTH_ACCESS_CONTROL_BLOCKED_EMAIL_DOMAINS             | Comma-separated list of email domains that cannot register.                                                                                                                                                                             |                              |
| AUTH_PASSWORD_MIN_LENGTH                              | Minimum password length.                                                                                                                                                                                                                | `3`                          |
| AUTH_PASSWORD_HIBP_ENABLED                            | User's password is checked against [Pwned Passwords](https://haveibeenpwned.com/Passwords).                                                                                                                                             | `false`                      |
| AUTH_USER_DEFAULT_ROLE                                | Default user role for registered users.                                                                                                                                                                                                 | `user`                       |
| AUTH_USER_DEFAULT_ALLOWED_ROLES                       | Comma-separated list of default allowed user roles.                                                                                                                                                                                     | `me,$AUTH_USER_DEFAULT_ROLE` |
| AUTH_LOCALE_DEFAULT                                   |                                                                                                                                                                                                                                         | `en`                         |
| AUTH_LOCALE_ALLOWED_LOCALES                           |                                                                                                                                                                                                                                         | `en`                         |
| AUTH_LOG_LEVEL                                        | Define the log level of the application:. Accepted values: `debug`, `info`, `warn`, `error`, `silent`.                                                                                                                                  | `info`                       |
| AUTH_EMAIL_PASSWORDLESS_ENABLED                       | Enables passwordless authentication by email. The SMTP server must then be configured.                                                                                                                                                  | `false`                      |
| AUTH_SIGNIN_MASTER_PASSWORD                           | Password that works to connect to any account (for debugging purposes)                                                                                                                                                                  | null                         |
| AUTH_SMS_PASSWORDLESS_ENABLED                         | Enables passwordless authentication by SMS. An SMS provider must then be configured.                                                                                                                                                    | `false`                      |
| AUTH_SMS_PROVIDER                                     | SMS provider name. Only `twilio` is possible as an option for now.                                                                                                                                                                      |                              |
| AUTH_SMS_TEST_PHONE_NUMBERS                           | Comma separated list of test phone numbers which can be used without any provider set. **The verification code can be found in the logs upon sign in**.                                                                                 |                              |
| AUTH_SMS_TWILIO_ACCOUNT_SID                           |                                                                                                                                                                                                                                         |                              |
| AUTH_SMS_TWILIO_AUTH_TOKEN                            |                                                                                                                                                                                                                                         |                              |
| AUTH_SMS_TWILIO_MESSAGING_SERVICE_ID                  |                                                                                                                                                                                                                                         |                              |
| AUTH_EMAIL_SIGNIN_EMAIL_VERIFIED_REQUIRED             | When enabled, any email-based authentication requires emails to be verified by a link sent to this email.                                                                                                                               | `true`                       |
| AUTH_ACCESS_CONTROL_ALLOWED_REDIRECT_URLS             | Comma-separated list of allowed redirect URLs that can be passed on as an option. Any sub-path will be considered valid. Supports wildcards and other [micromatch patterns](https://github.com/micromatch/micromatch#matching-features) |                              |
| AUTH_MFA_ENABLED                                      | Enables users to use Multi Factor Authentication.                                                                                                                                                                                       | `false`                      |
| AUTH_MFA_TOTP_ISSUER                                  | The name of the One Time Password (OTP) issuer. Probably your app's name.                                                                                                                                                               | `hasura-auth`                |
| AUTH_ACCESS_TOKEN_EXPIRES_IN                          | Number of seconds before the access token (JWT) expires.                                                                                                                                                                                | `900`(15 minutes)            |
| AUTH_REFRESH_TOKEN_EXPIRES_IN                         | Number of seconds before the refresh token expires.                                                                                                                                                                                     | `2592000` (30 days)          |
| AUTH_JWT_CUSTOM_CLAIMS                                |                                                                                                                                                                                                                                         |                              |
| AUTH_WEBAUTHN_ENABLED                                 | When enabled, passwordless Webauthn authentication can be done via device supported strong authenticators like fingerprint, Face ID, etc.                                                                                               | false                        |
| AUTH_WEBAUTHN_RP_NAME                                 | Relying party name. Friendly name visual to the user informing who requires the authentication. Probably your app's name.                                                                                                               |                              |
| AUTH_WEBAUTHN_RP_ORIGINS                              | Array of URLs where the registration is permitted and should have occurred on. `AUTH_CLIENT_URL` will be automatically added to the list of origins if is set.                                                                          |                              |
| AUTH_WEBAUTHN_ATTESTATION_TIMEOUT                     | How long (in ms) the user can take to complete authentication.                                                                                                                                                                          | `60000` (1 minute)           |

# OAuth environment variables

| Name (a star<b>\*</b> means the variable is required when the provider is enabled) | Default value                       |
| ---------------------------------------------------------------------------------- | ----------------------------------- |
| AUTH_PROVIDER_GITHUB_ENABLED                                                       | `false`                             |
| AUTH_PROVIDER_GITHUB_CLIENT_ID<b>\*</b>                                            |                                     |
| AUTH_PROVIDER_GITHUB_CLIENT_SECRET<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_GITHUB_AUTHORIZATION_URL                                             |                                     |
| AUTH_PROVIDER_GITHUB_TOKEN_URL                                                     |                                     |
| AUTH_PROVIDER_GITHUB_USER_PROFILE_URL                                              |                                     |
| AUTH_PROVIDER_GITHUB_SCOPE                                                         | `user:email `                       |
| AUTH_PROVIDER_GOOGLE_ENABLED                                                       | `false`                             |
| AUTH_PROVIDER_GOOGLE_CLIENT_ID<b>\*</b>                                            |                                     |
| AUTH_PROVIDER_GOOGLE_CLIENT_SECRET<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_GOOGLE_SCOPE                                                         | `email,profile`                     |
| AUTH_PROVIDER_FACEBOOK_ENABLED                                                     | `false`                             |
| AUTH_PROVIDER_FACEBOOK_CLIENT_ID<b>\*</b>                                          |                                     |
| AUTH_PROVIDER_FACEBOOK_CLIENT_SECRET<b>\*</b>                                      |                                     |
| AUTH_PROVIDER_FACEBOOK_PROFILE_FIELDS                                              | `email,photos,displayName`          |
| AUTH_PROVIDER_FACEBOOK_SCOPE                                                       | `email`                             |
| AUTH_PROVIDER_TWITTER_ENABLED                                                      | `false`                             |
| AUTH_PROVIDER_TWITTER_CONSUMER_KEY<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_TWITTER_CONSUMER_SECRET<b>\*</b>                                     |                                     |
| AUTH_PROVIDER_LINKEDIN_ENABLED                                                     |                                     |
| AUTH_PROVIDER_LINKEDIN_CLIENT_ID<b>\*</b>                                          |                                     |
| AUTH_PROVIDER_LINKEDIN_CLIENT_SECRET<b>\*</b>                                      |                                     |
| AUTH_PROVIDER_LINKEDIN_SCOPE                                                       | `r_emailaddress,r_liteprofile`      |
| AUTH_PROVIDER_APPLE_ENABLED                                                        | `false`                             |
| AUTH_PROVIDER_APPLE_CLIENT_ID<b>\*</b>                                             |                                     |
| AUTH_PROVIDER_APPLE_TEAM_ID<b>\*</b>                                               |                                     |
| AUTH_PROVIDER_APPLE_KEY_ID<b>\*</b>                                                |                                     |
| AUTH_PROVIDER_APPLE_PRIVATE_KEY<b>\*</b>                                           | Base64 format                       |
| AUTH_PROVIDER_APPLE_SCOPE                                                          | `name,email`                        |
| AUTH_PROVIDER_WINDOWS_LIVE_ENABLED                                                 | `false`                             |
| AUTH_PROVIDER_WINDOWS_LIVE_CLIENT_ID<b>\*</b>                                      |                                     |
| AUTH_PROVIDER_WINDOWS_LIVE_CLIENT_SECRET<b>\*</b>                                  |                                     |
| AUTH_PROVIDER_WINDOWS_LIVE_SCOPE                                                   | `wl.basic,wl.emails`                |
| AUTH_PROVIDER_SPOTIFY_ENABLED                                                      | `false`                             |
| AUTH_PROVIDER_SPOTIFY_CLIENT_ID<b>\*</b>                                           |                                     |
| AUTH_PROVIDER_SPOTIFY_CLIENT_SECRET<b>\*</b>                                       |                                     |
| AUTH_PROVIDER_SPOTIFY_SCOPE                                                        | `user-read-email,user-read-private` |
| AUTH_PROVIDER_GITLAB_ENABLED                                                       | `false`                             |
| AUTH_PROVIDER_GITLAB_CLIENT_ID<b>\*</b>                                            |                                     |
| AUTH_PROVIDER_GITLAB_CLIENT_SECRET<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_GITLAB_BASE_URL                                                      |                                     |
| AUTH_PROVIDER_GITLAB_SCOPE                                                         | `read_user`                         |
| AUTH_PROVIDER_BITBUCKET_ENABLED                                                    | `false`                             |
| AUTH_PROVIDER_BITBUCKET_CLIENT_ID<b>\*</b>                                         |                                     |
| AUTH_PROVIDER_BITBUCKET_CLIENT_SECRET<b>\*</b>                                     |                                     |
| AUTH_PROVIDER_STRAVA_ENABLED                                                       | `false`                             |
| AUTH_PROVIDER_STRAVA_CLIENT_ID<b>\*</b>                                            |                                     |
| AUTH_PROVIDER_STRAVA_CLIENT_SECRET<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_STRAVA_SCOPE                                                         | `profile:read_all`                  |
| AUTH_PROVIDER_WORKOS_ENABLED                                                       | `false`                             |
| AUTH_PROVIDER_WORKOS_CLIENT_ID<b>\*</b>                                            |                                     |
| AUTH_PROVIDER_WORKOS_CLIENT_SECRET<b>\*</b>                                        |                                     |
| AUTH_PROVIDER_WORKOKS_DEFAULT_DOMAIN                                               |                                     |
| AUTH_PROVIDER_WORKOS_DEFAULT_ORGANIZATION                                          |                                     |
| AUTH_PROVIDER_WORKOS_DEFAULT_CONNECTION                                            |                                     |
| AUTH_PROVIDER_AZUREAD_ENABLED                                                      |                                     |
| AUTH_PROVIDER_AZUREAD_CLIENT_ID                                                    |                                     |
| AUTH_PROVIDER_AZUREAD_CLIENT_SECRET                                                |                                     |
| AUTH_PROVIDER_AZUREAD_TENANT                                                       |                                     |


src/utils/env.ts
import { logger } from '@/logger';
import { JwtSecret } from '@/types';
import {
  castBooleanEnv,
  castIntEnv,
  castObjectEnv,
  castStringArrayEnv,
  castStringEnv,
} from '../config/utils';

export const ENV = {
  get PWD() {
    return castStringEnv('PWD', '/');
  },

  // HASURA
  get HASURA_GRAPHQL_JWT_SECRET() {
    return castObjectEnv<JwtSecret>('HASURA_GRAPHQL_JWT_SECRET');
  },
  get HASURA_GRAPHQL_DATABASE_URL() {
    return castStringEnv('HASURA_GRAPHQL_DATABASE_URL');
  },
  get HASURA_GRAPHQL_ADMIN_SECRET() {
    return castStringEnv('HASURA_GRAPHQL_ADMIN_SECRET');
  },
  get HASURA_GRAPHQL_GRAPHQL_URL() {
    return castStringEnv('HASURA_GRAPHQL_GRAPHQL_URL');
  },

  // SERVER
  get AUTH_PORT() {
    return castIntEnv('AUTH_PORT', 4000);
  },
  get AUTH_SERVER_URL() {
    return castStringEnv('AUTH_SERVER_URL');
  },

  // SMTP
  get AUTH_SMTP_PASS() {
    return castStringEnv('AUTH_SMTP_PASS');
  },
  get AUTH_SMTP_HOST() {
    return castStringEnv('AUTH_SMTP_HOST');
  },
  get AUTH_SMTP_USER() {
    return castStringEnv('AUTH_SMTP_USER');
  },
  get AUTH_SMTP_SENDER() {
    return castStringEnv('AUTH_SMTP_SENDER');
  },
  get AUTH_SMTP_AUTH_METHOD() {
    return castStringEnv('AUTH_SMTP_AUTH_METHOD', 'PLAIN');
  },
  get AUTH_SMTP_PORT() {
    return castIntEnv('AUTH_SMTP_PORT', 587);
  },
  get AUTH_SMTP_SECURE() {
    // note: false disables SSL (deprecated)
    return castBooleanEnv('AUTH_SMTP_SECURE', false);
  },

  // SMS
  get AUTH_SMS_PROVIDER() {
    return castStringEnv('AUTH_SMS_PROVIDER');
  },
  get AUTH_SMS_TEST_PHONE_NUMBERS() {
    return castStringArrayEnv('AUTH_SMS_TEST_PHONE_NUMBERS', []);
  },

  // TWILIO
  get AUTH_SMS_TWILIO_ACCOUNT_SID() {
    return castStringEnv('AUTH_SMS_TWILIO_ACCOUNT_SID');
  },
  get AUTH_SMS_TWILIO_AUTH_TOKEN() {
    return castStringEnv('AUTH_SMS_TWILIO_AUTH_TOKEN');
  },
  get AUTH_SMS_TWILIO_MESSAGING_SERVICE_ID() {
    return castStringEnv('AUTH_SMS_TWILIO_MESSAGING_SERVICE_ID');
  },

  // GRAVATAR
  get AUTH_GRAVATAR_ENABLED() {
    return castBooleanEnv('AUTH_GRAVATAR_ENABLED', true);
  },
  get AUTH_GRAVATAR_DEFAULT() {
    return castStringEnv('AUTH_GRAVATAR_DEFAULT', 'blank');
  },
  get AUTH_GRAVATAR_RATING() {
    return castStringEnv('AUTH_GRAVATAR_RATING', 'g');
  },

  // CLIENT
  get AUTH_CLIENT_URL() {
    return castStringEnv('AUTH_CLIENT_URL').toLocaleLowerCase();
  },

  // WEBAUTHN
  get AUTH_WEBAUTHN_ENABLED() {
    return castBooleanEnv('AUTH_WEBAUTHN_ENABLED', false);
  },
  get AUTH_WEBAUTHN_RP_NAME() {
    return castStringEnv('AUTH_WEBAUTHN_RP_NAME', '');
  },
  get AUTH_WEBAUTHN_RP_ORIGINS() {
    const origins = castStringArrayEnv('AUTH_WEBAUTHN_RP_ORIGINS', []);
    const clientUrl = ENV.AUTH_CLIENT_URL;
    if (clientUrl) {
      origins.push(clientUrl);
    }
    return origins.filter(Boolean);
  },
  get AUTH_WEBAUTHN_ATTESTATION_TIMEOUT() {
    return castIntEnv('AUTH_WEBAUTHN_ATTESTATION_TIMEOUT', 60000);
  },

  // RECAPTCHA
  get AUTH_RECAPTCHA_SITE_KEY() {
    return castStringEnv('AUTH_RECAPTCHA_SITE_KEY', '');
  },
  get AUTH_RECAPTCHA_GOOGLE_API_KEY() {
    return castStringEnv('AUTH_RECAPTCHA_GOOGLE_API_KEY', '');
  },
  get AUTH_RECAPTCHA_PROJECT_ID() {
    return castStringEnv('AUTH_RECAPTCHA_PROJECT_ID', '');
  },

  // SIGN UP
  get AUTH_ANONYMOUS_USERS_ENABLED() {
    return castBooleanEnv('AUTH_ANONYMOUS_USERS_ENABLED', false);
  },
  get AUTH_DISABLE_NEW_USERS() {
    return castBooleanEnv('AUTH_DISABLE_NEW_USERS', false);
  },
  get AUTH_ACCESS_CONTROL_ALLOWED_EMAILS() {
    return castStringArrayEnv('AUTH_ACCESS_CONTROL_ALLOWED_EMAILS', []);
  },
  get AUTH_ACCESS_CONTROL_ALLOWED_EMAIL_DOMAINS() {
    return castStringArrayEnv('AUTH_ACCESS_CONTROL_ALLOWED_EMAIL_DOMAINS', []);
  },
  get AUTH_ACCESS_CONTROL_BLOCKED_EMAILS() {
    return castStringArrayEnv('AUTH_ACCESS_CONTROL_BLOCKED_EMAILS', []);
  },
  get AUTH_ACCESS_CONTROL_BLOCKED_EMAIL_DOMAINS() {
    return castStringArrayEnv('AUTH_ACCESS_CONTROL_BLOCKED_EMAIL_DOMAINS', []);
  },
  get AUTH_PASSWORD_MIN_LENGTH() {
    return castIntEnv('AUTH_PASSWORD_MIN_LENGTH', 3);
  },
  get AUTH_PASSWORD_HIBP_ENABLED() {
    return castBooleanEnv('AUTH_PASSWORD_HIBP_ENABLED', false);
  },
  get AUTH_USER_DEFAULT_ROLE() {
    return castStringEnv('AUTH_USER_DEFAULT_ROLE', 'user');
  },
  get AUTH_USER_DEFAULT_ALLOWED_ROLES() {
    const allowedRoles = [
      'me',
      this.AUTH_USER_DEFAULT_ROLE,
      ...castStringArrayEnv('AUTH_USER_DEFAULT_ALLOWED_ROLES'),
    ];
    // remove duplicates
    return [...new Set(allowedRoles)];
  },
  get AUTH_LOCALE_DEFAULT() {
    return castStringEnv('AUTH_LOCALE_DEFAULT', 'en');
  },
  get AUTH_LOCALE_ALLOWED_LOCALES() {
    const locales = [
      ENV.AUTH_LOCALE_DEFAULT,
      ...castStringArrayEnv('AUTH_LOCALE_ALLOWED_LOCALES'),
    ];
    // remove duplicates
    return [...new Set(locales)];
  },
  get AUTH_SIGNUP_RECAPTCHA_CHALLENGE() {
    return castBooleanEnv('AUTH_SIGNUP_RECAPTCHA_CHALLENGE', false);
  },

  // SIGN IN
  get AUTH_EMAIL_PASSWORDLESS_ENABLED() {
    return castBooleanEnv('AUTH_EMAIL_PASSWORDLESS_ENABLED', false);
  },
  get AUTH_SMS_PASSWORDLESS_ENABLED() {
    return castBooleanEnv('AUTH_SMS_PASSWORDLESS_ENABLED', false);
  },
  get AUTH_EMAIL_SIGNIN_EMAIL_VERIFIED_REQUIRED() {
    return castBooleanEnv('AUTH_EMAIL_SIGNIN_EMAIL_VERIFIED_REQUIRED', true);
  },
  get AUTH_SIGNIN_RECAPTCHA_CHALLENGE() {
    return castBooleanEnv('AUTH_SIGNIN_RECAPTCHA_CHALLENGE', false);
  },
  get AUTH_SIGNIN_MASTER_PASSWORD() {
    return castStringEnv('AUTH_SIGNIN_MASTER_PASSWORD');
  },

  // get AUTH_SIGNIN_PHONE_NUMBER_VERIFIED_REQUIRED() {
  //   return castBooleanEnv('AUTH_SIGNIN_PHONE_NUMBER_VERIFIED_REQUIRED', true);
  // },
  get AUTH_ACCESS_CONTROL_ALLOWED_REDIRECT_URLS() {
    return castStringArrayEnv('AUTH_ACCESS_CONTROL_ALLOWED_REDIRECT_URLS').map(
      (v) => v.toLowerCase()
    );
  },
  get AUTH_MFA_ENABLED() {
    return castBooleanEnv('AUTH_MFA_ENABLED', false);
  },
  get AUTH_MFA_TOTP_ISSUER() {
    return castStringEnv('AUTH_MFA_TOTP_ISSUER', 'hasura-auth');
  },

  // TOKENS
  get AUTH_ACCESS_TOKEN_EXPIRES_IN() {
    return castIntEnv('AUTH_ACCESS_TOKEN_EXPIRES_IN', 900);
  },
  get AUTH_REFRESH_TOKEN_EXPIRES_IN() {
    return castIntEnv('AUTH_REFRESH_TOKEN_EXPIRES_IN', 2_592_000);
  },

  // EMAIL TEMPLATES
  get AUTH_EMAIL_TEMPLATE_FETCH_URL() {
    return castStringEnv('AUTH_EMAIL_TEMPLATE_FETCH_URL');
  },

  get AUTH_JWT_CUSTOM_CLAIMS() {
    try {
      return castObjectEnv<Record<string, string>>('AUTH_JWT_CUSTOM_CLAIMS');
    } catch {
      logger.warn(
        'AUTH_JWT_CUSTOM_CLAIMS cannot be parsed. Will ignore custom claims.'
      );
      return {};
    }
  },

  get AUTH_USER_SESSION_VARIABLE_FIELDS(): Record<string, string> {
    return this.AUTH_USER_SESSION_VARIABLE_FIELDS;
  },

  get AUTH_CONCEAL_ERRORS() {
    return castBooleanEnv('AUTH_CONCEAL_ERRORS', false);
  },
  // * See ../server.ts
  // get AUTH_SKIP_INIT() {
  //   return castBooleanEnv('AUTH_SKIP_INIT', false);
  // },
  // get AUTH_SKIP_SERVE() {
  //   return castBooleanEnv('AUTH_SKIP_SERVE', false);
  // },
};
